### PR TITLE
c++ API ( average_accumulates, tensor_array_to_tensor and average_accumulates) error message enhance. test=develop

### PR DESCRIPTION
**Pull request by chenjiaoAngel**
c++ API接口错误信息增强：
average_accumulates_op 
tensor_array_to_tensor_op
average_accumulates_op

### CODE REVIEW

File: paddle/fluid/operators/average_accumulates_op.cc [hunk 1]
```diff
@@ -48,47 +48,64 @@ class AverageAccumulatesOp : public framework::OperatorWithKernel {
...
-    PADDLE_ENFORCE(ctx->HasOutput("out_num_accumulates"),
-                   "Output (num_accumulates) of average_accumulates op should "
-                   "not be null.");
-    PADDLE_ENFORCE(ctx->HasOutput("out_old_num_accumulates"),
-                   "Output (old_num_accumulates) of average_accumulates op "
-                   "should not be null.");
-    PADDLE_ENFORCE(
-        ctx->HasOutput("out_num_updates"),
-        "Output (num_updates) of average_accumulates op should not be null.");
+    PADDLE_ENFORCE_EQ(
```
Comment (GaoWei8) [inline paddle/fluid/operators/average_accumulates_op.cc]: 建议使用最新输入检查标准。
d80c27de84c1e0ca739a7c9222a32fa9

Comment (chenjiaoAngel) [reply paddle/fluid/operators/average_accumulates_op.cc]: hao

File: paddle/fluid/operators/get_tensor_from_selected_rows_op.cc [hunk 2]
```diff
@@ -23,21 +23,25 @@ class GetTensorFromSelectedRowsOp : public framework::OperatorWithKernel {
...
+    OP_INOUT_CHECK(ctx->HasOutput("Out"), "Output", "Out",
+                   "GetTensorFromSelectedRows");
+
+    PADDLE_ENFORCE_EQ(ctx->GetInputsVarType("X").front() ==
+                          framework::proto::VarType::SELECTED_ROWS,
+                      true, platform::errors::InvalidArgument(
+                                "The input X's type should be SelectedRows, "
+                                "but the received is %s",
+                                ctx->Inputs("X").front(),
+                                ctx->GetInputsVarType("X").front()));
```
Comment (zhupengyang) [inline paddle/fluid/operators/get_tensor_from_selected_rows_op.cc]:
```
PADDLE_ENFORCE_EQ(ctx->GetInputsVarType("X").front(), framework::proto::VarType::SELECTED_ROWS, platform::errors::InvalidArgument("xxx"))
```
其它 PADDLE_ENFORCE_EQ 类似，比较两个变量，而不是变量相等和true比较

File: paddle/fluid/operators/get_tensor_from_selected_rows_op.cc [hunk 3]
```diff
@@ -23,21 +23,25 @@ class GetTensorFromSelectedRowsOp : public framework::OperatorWithKernel {
...
+    PADDLE_ENFORCE_EQ(ctx->GetOutputsVarType("Out").front() ==
```
Comment (zhupengyang) [inline paddle/fluid/operators/get_tensor_from_selected_rows_op.cc]: please fix it

File: paddle/fluid/operators/tensor_array_to_tensor_op.cc [hunk 4]
```diff
@@ -96,7 +96,8 @@ class LoDTensorArray2TensorOp : public framework::OperatorBase {
         *scope.FindVar(Output("OutIndex"))->GetMutable<framework::LoDTensor>();
 
     const size_t n = inx.size();
-    PADDLE_ENFORCE_GT(n, 0, "Input tensorarray size should > 0.");
+    PADDLE_ENFORCE_GT(n, 0, platform::errors::InvalidArgument(
+                                "Input tensorarray size should > 0."));
```
Comment (zhupengyang) [inline paddle/fluid/operators/tensor_array_to_tensor_op.cc]: 报错信息里面完善一下，增加 ```"But received n: %d", n```

File: paddle/fluid/operators/tensor_array_to_tensor_op.cc [hunk 5]
```diff
@@ -232,7 +233,8 @@ class LoDTensorArray2TensorGradOp : public framework::OperatorBase {
 
     auto &inx = scope.FindVar(Input("X"))->Get<framework::LoDTensorArray>();
     const size_t n = inx.size();
-    PADDLE_ENFORCE_GT(n, 0, "Input tensorarray size should > 0.");
+    PADDLE_ENFORCE_GT(n, 0, platform::errors::InvalidArgument(
+                                "Input tensorarray size should > 0."));
```
Comment (zhupengyang) [inline paddle/fluid/operators/tensor_array_to_tensor_op.cc]: please enhance it

File: paddle/fluid/operators/get_tensor_from_selected_rows_op.cc [hunk 6]
```diff
@@ -23,21 +23,25 @@ class GetTensorFromSelectedRowsOp : public framework::OperatorWithKernel {
...
+    OP_INOUT_CHECK(ctx->HasOutput("Out"), "Output", "Out",
+                   "GetTensorFromSelectedRows");
+
+    PADDLE_ENFORCE_EQ(
+        ctx->GetInputsVarType("X").front(),
+        framework::proto::VarType::SELECTED_ROWS,
+        platform::errors::InvalidArgument(
+            "The input X's type should be SelectedRows, "
+            "but the received is %s",
+            ctx->Inputs("X").front(), ctx->GetInputsVarType("X").front()));
```
Comment (zhupengyang) [inline paddle/fluid/operators/get_tensor_from_selected_rows_op.cc]: ```The input X's type should be SelectedRows,``` --> ```The input X(%s)'s type should be SelectedRows,```

File: paddle/fluid/operators/get_tensor_from_selected_rows_op.cc [hunk 7]
```diff
@@ -23,21 +23,25 @@ class GetTensorFromSelectedRowsOp : public framework::OperatorWithKernel {
...
+    PADDLE_ENFORCE_EQ(ctx->GetOutputsVarType("Out").front(),
+                      framework::proto::VarType::LOD_TENSOR,
+                      platform::errors::InvalidArgument(
+                          "The output Out's type should be LoDTensor, "
+                          "but the received is %s",
+                          ctx->Outputs("Out").front(),
+                          ctx->GetOutputsVarType("Out").front()));
```
Comment (zhupengyang) [inline paddle/fluid/operators/get_tensor_from_selected_rows_op.cc]: 同上

**Review by zhupengyang — APPROVED** [review]
LGTM

**Review by hong19860320 — APPROVED** [review]
LGTM

**Pull request merged**